dbms/src/DataStreams/NullableAdapterBlockInputStream.cpp

```diff
@@ -102,8 +102,8 @@ void NullableAdapterBlockInputStream::buildActions(
 
     for (size_t i = 0; i < in_size; ++i)
     {
-        const auto & in_elem  = in_sample.getByPosition(i);
-        const auto & out_elem = out_sample.getByPosition(i);
+        const auto & in_elem  = in_sample.safeGetByPosition(i);
```

Comment:
Don't we should give better exception message like "Unmatched number of columns in INSERT SELECT"?